src/artemis/parameters/plan_specific/grid_scan_with_edge_detect_params.py

```diff
@@ -18,7 +17,7 @@
 
 
 @dataclass
-class GridScanWithEdgeDetectParams(DataClassJsonMixin, AbstractExperimentParameterBase):
+class GridScanWithEdgeDetectParams(AbstractExperimentParameterBase):
```

Comment:
Query: `RotationScanParams` inherits from `BaseModel`. Do we have a preference on when to do that and when to use `pydantic.dataclass`? From a google I think maybe `BaseModel` gives you more? So I would suggest we standardise on them all using `BaseModel`? This may be a different ticket...

Reply:
Yes, maybe it's better to use models for all of them, I just changed the ones which were still dataclass because it was easier

Reply:
Cool, I'll make an issue